pika/amqp_object.py

```diff
@@ -21,6 +21,12 @@ def __repr__(self):
             return "<%s>" % self.NAME
         return "<%s(%s)>" % (self.NAME, sorted(items))
 
+    def __eq__(self, other):
```

Comment:
How about adding this here:

```
if self is other:
    return True
```

Reply:
if self is other then the comparison of the dicts point to the same dict object. hence there should not be much of a difference.

Reply:
`self.__dict__ == other.__dict__` is smart enough to do a reference comparison first, then? That makes sense.